File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        if len(nums) == 2:
            if nums[0] > nums[1]:
                return nums[1]
            return nums[0]

        middle_index = len(nums) // 2
        middle_number = nums[middle_index]
        last_number = nums[-1]

        if middle_number > last_number:
            return self.findMin(nums[middle_index:])
        else:
            return self.findMin(nums[:middle_index + 1])

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] arr) {
        //go for the sorted part first, 
        // pick the lowest from there 
        // and you get the minimum 
        // then find in the rotated part
     int low= 0 , high = arr.length - 1;
     while (low < high){
         int mid = low + (high - low) / 2;
    if(arr[mid] > arr[high]){
        low = mid + 1; // Minimum is in the right half
     }
    else {
        high = mid; // Minimum is in the left half or at mid
     }

    }
        return arr[low]; // The minimum element is at index low
    }
}